build/yaml-cpp.m4

```diff
@@ -74,7 +74,7 @@ fi
 ],
 [
   has_yaml_cpp=no
-  TS_ADDTO(CPPFLAGS, [-I\${abs_top_srcdir}/lib/yamlcpp/include])
+  TS_ADDTO(CPPFLAGS, [-fPIC -I\${abs_top_srcdir}/lib/yamlcpp/include])
```

Comment:
This shouldn't been needed.

Reply:
I added this because the build fails when we include the yamlcpp to some code in lib/ts 
 showing 
'''
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(exceptions.o): relocation R_X86_64_32S against symbol `_ZTVN4YAML9ExceptionE' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(node_data.o): relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(parse.o): relocation R_X86_64_32 against symbol `__pthread_key_create@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(parser.o): relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(scanner.o): relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(scantoken.o): relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(simplekey.o): relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(singledocparser.o): relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(stream.o): relocation R_X86_64_32 against `.data' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(tag.o): relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(directives.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(memory.o): relocation R_X86_64_32 against symbol `__pthread_key_create@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(nodebuilder.o): relocation R_X86_64_32S against symbol `_ZTVN4YAML11NodeBuilderE' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(null.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(regex_yaml.o): relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(scanscalar.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(scantag.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /sharetchi/trafficserver/lib/yamlcpp/libyamlcpp.a(exp.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Nonrepresentable section on output
'''

Reply:
I don't see an option unless we compile yamlcpp as a dynamic lib.